File: android/app/src/main/kotlin/com/example/flutter_interactive/MyMethodChannel.kt
package com.example.flutter_interactive

import android.util.Log
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.MethodCall


class MyMethodChannel(messenger: BinaryMessenger) : MethodChannel.MethodCallHandler {

    private var myChannel: MethodChannel = MethodChannel(messenger, "com.flutter.test.channel")

    init {
        myChannel.setMethodCallHandler(this)
    }

    override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
        Log.e("call.method ", call.method)
        if (call.method == "TestData") {
            val name = call.argument("name") as String?
            val age = call.argument("age") as String?
            var map = mapOf("name" to "hello,$name",
                    "age" to "$age"
            )
            result.success(map)
        } else if (call.method == "native") {
            val name = "原生Android " + myChannel.javaClass.name
            myChannel.invokeMethod("fromNative", name)

        }
    }
}

File: ios/Runner/MyMethodChannel.swift
//
//  MyMethodChannel.swift
//  Runner
//
//  Created by 福建连 on 7/22/21.
//

import Flutter
import UIKit

public class MyMethodChannel {
    
    var channel:FlutterMethodChannel
    
    init(messenger: FlutterBinaryMessenger) {
        channel = FlutterMethodChannel(name: "com.flutter.test.channel", binaryMessenger: messenger)
        channel.setMethodCallHandler { (call:FlutterMethodCall, result:@escaping FlutterResult) in
            if (call.method == "TestData") {
                if let dict = call.arguments as? Dictionary<String, Any> {
                    let name:String = dict["name"] as? String ?? ""
                    let age:String = dict["age"] as? String ?? ""
                    result(["name":"hello,\(name)","age":age])
                }
            }else if (call.method == "native") {
                let data = "原生ios " + String(cString:object_getClassName(self.channel))
                self.channel.invokeMethod("fromNative", arguments:data)
            }
        }
    }
}
